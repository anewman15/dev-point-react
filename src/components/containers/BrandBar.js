/* eslint-disable react/self-closing-comp */
import { Link } from 'react-router-dom';

function BrandBar() {
  const handleClick = () => {
    const sideNav = document.getElementById('side-nav');
    sideNav.classList.toggle('is-active');
  };

  return (
    <nav className="navbar is-fixed-top" role="navigation" aria-label="main navigation">
      <div className="navbar-brand mx-3 my-3 px-2 is-text-centered">
        <Link className="" to="/">
          <img src={`${process.env.PUBLIC_URL}/logo512.png`} alt="dev-lookup-logo" width="60" height="60" />
        </Link>

        <a
          href="#side-nav"
          role="button"
          className="navbar-burger"
          aria-label="menu"
          aria-expanded="false"
          data-target="side-nav"
          onClick={handleClick}
        >
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
          <span aria-hidden="true"></span>
        </a>
      </div>
    </nav>
  );
}

export default BrandBar;
