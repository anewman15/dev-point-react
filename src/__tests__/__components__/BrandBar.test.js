import { unmountComponentAtNode } from 'react-dom';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import BrandBar from '../../components/containers/BrandBar';

let container = null;
beforeEach(() => {
  container = document.createElement('div');
  document.body.appendChild(container);
});

afterEach(() => {
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});

describe('BrandBar', () => {
  it('renders the BrandBar component to the dom', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <BrandBar />
        </MemoryRouter>,
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
