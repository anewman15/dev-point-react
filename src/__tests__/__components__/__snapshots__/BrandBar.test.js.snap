// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrandBar renders the BrandBar component to the dom 1`] = `
<nav
  aria-label="main navigation"
  className="navbar is-fixed-top"
  role="navigation"
>
  <div
    className="navbar-brand mx-3 my-3 px-2 is-text-centered"
  >
    <a
      className=""
      href="/"
      onClick={[Function]}
    >
      <img
        alt="dev-lookup-logo"
        height="60"
        src="/logo512.png"
        width="60"
      />
    </a>
    <a
      aria-expanded="false"
      aria-label="menu"
      className="navbar-burger"
      data-target="side-nav"
      href="#side-nav"
      onClick={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
</nav>
`;
